File: sdks/js/src/core/level.ts
/** Log levels matching the LOXA spec. */
export const LevelDebug = 0;
export const LevelInfo = 1;
export const LevelNotice = 2;
export const LevelWarn = 3;
export const LevelError = 4;
export const LevelFatal = 5;

export type Level = typeof LevelDebug | typeof LevelInfo | typeof LevelNotice | typeof LevelWarn | typeof LevelError | typeof LevelFatal;

const LEVEL_NAMES: Record<number, string> = {
  [LevelDebug]: 'debug',
  [LevelInfo]: 'info',
  [LevelNotice]: 'notice',
  [LevelWarn]: 'warn',
  [LevelError]: 'error',
  [LevelFatal]: 'fatal',
};

export function parseLevel(s: string): Level {
  const lower = s.toLowerCase();
  for (const [k, v] of Object.entries(LEVEL_NAMES)) {
    if (v === lower) return Number(k) as Level;
  }
  return LevelInfo;
}

export function levelName(l: Level): string {
  return LEVEL_NAMES[l] ?? 'info';
}

File: sdks/go/src/core/level.go
package core

import "strings"

// Level represents the severity level of a log event.
type Level uint8

const (
	LevelDebug Level = iota
	LevelInfo
	LevelNotice
	LevelWarn
	LevelError
	LevelFatal
)

// String returns the lower-case string representation of the level.
func (l Level) String() string {
	switch l {
	case LevelDebug:
		return "debug"
	case LevelInfo:
		return "info"
	case LevelNotice:
		return "notice"
	case LevelWarn:
		return "warn"
	case LevelError:
		return "error"
	case LevelFatal:
		return "fatal"
	default:
		return "info"
	}
}

// ParseLevel parses a string into a Level. Defaults to LevelInfo on unknown input.
func ParseLevel(s string) Level {
	switch strings.ToLower(strings.TrimSpace(s)) {
	case "debug":
		return LevelDebug
	case "info":
		return LevelInfo
	case "notice":
		return LevelNotice
	case "warn", "warning":
		return LevelWarn
	case "error":
		return LevelError
	case "fatal":
		return LevelFatal
	default:
		return LevelInfo
	}
}

// Parse parses a string into a Level. Defaults to LevelInfo on unknown input.
func Parse(s string) Level {
	return ParseLevel(s)
}
